tests/aws/services/cloudformation/engine/test_references.py

```diff
@@ -53,3 +55,42 @@ def test_fn_sub_cases(self, deploy_cfn_template, aws_client, snapshot):
         )
 
         snapshot.match("outputs", deployment.outputs)
+
+
+@markers.aws.only_localstack
+def test_useful_error_when_invalid_ref(deploy_cfn_template):
```

Comment:
Good idea to add a test here tp avoid us messing it up when refactoring on the future. We might want to pull out tests like this from the file in the future though.